uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -314,6 +316,21 @@ define([
     }
 
     /**
+     * Initialize the original opening balance when in edit mode. This allows
+     * to display the opening balance before the request was created. The
+     * formula is absence type reminder + balance change (balance change is a
+     * negative number so it needs to be subtracted).
+     */
+    function initOriginalOpeningBalance () {
+      if (vm.isMode('edit')) {
```

Comment:
Original Opening Balance, should be saved for approved requests in edit mode only, right? Then you are not checking if it is approved request only.

Reply:
Sorry, my bad, it's not just for open leave request, it's for any leave request that has already been saved.

Reply:
I think the requirement for the ticket is applicable for approved requests only, because only when a request is approved, the balance gets updated in the backend. Doing the same for non approved requests, will be wrong, is not it?
Suppose you have entitlement for 20 days, and a leave request for 10 days has been created, but till it is approved, the `selectedAbsenceType.remainder` still remains 20. So if you show 20+10 =30, it will be wrong.

Reply:
in that case the remainder will be 10. I double checked, and even if the request is in an open state the remainder stores the balance as if the request is approved.

Reply:
Here's the *after* for a request in a open state:
![anim](https://user-images.githubusercontent.com/1642119/29826225-36f629e2-8c83-11e7-8ab6-48484f5b6d1f.gif)